src/dhtproto/client/mixins/NeoSupport.d

```diff
@@ -1097,6 +1097,165 @@ template NeoSupport ( )
 
             return res;
         }
+
+        /***********************************************************************
+
+            Struct to provide an opApply for a task-blocking GetChannels.
+
+        ***********************************************************************/
+
+        public struct GetChannelsFruct
+        {
+            import ocean.core.array.Mutation: copy;
+
+            /// User task to resume/suspend.
+            private Task task;
+
+            /// Name of the current channel.
+            private mstring* channel_name;
+
+            /// Possible states of the request.
+            private enum State
+            {
+                /// The request is running.
+                Running,
+
+                /// The user has stopped this request by breaking from foreach
+                /// (the request may still be running for some time, but all
+                /// channel names will be ignored).
+                Stopped,
+
+                /// The request has finished on all nodes.
+                Finished
+            }
+
+            /// Indicator of the request's state.
+            private State state;
+
+            /// Neo instance to assign the request with.
+            private Neo neo;
+
+            /// Error indicator.
+            public bool error;
+
+            /*******************************************************************
+
+                Notifier used to set the local values and resume the task.
+
+                Params:
+                    info = information and payload about the event user has
+                        been notified about
+                    args = arguments passed by the user when starting request
+
+            *******************************************************************/
+
+            private void notifier ( Neo.GetChannels.Notification info,
+                Neo.GetChannels.Args args )
+            {
+                with ( info.Active ) switch ( info.active )
+                {
+                    case received:
+                        // Ignore all received value on user break.
+                        if (this.state == State.Stopped)
+                            break;
+
+                        copy(*this.channel_name,
+                            cast(cstring)info.received.value);
+                        enableStomping(*this.channel_name);
```

Comment:
Ditto: I think `copy` does this already.

Reply:
At least in ocean v3.3.1 (used by dhtproto) it doesn't. https://github.com/sociomantic-tsunami/ocean/blob/v3.3.1/src/ocean/core/array/Mutation.d#L925-L933

(Ditto for v3.x.x.)

Reply:
The first argument here is `Buffer`, and `dest.length` does that internally: https://github.com/sociomantic-tsunami/ocean/blob/v3.x.x/src/ocean/core/Buffer.d#L140

Reply:
Ah indeed.